src/pygeoprocessing/geoprocessing.py

```diff
@@ -29,6 +29,23 @@
 import shapely.prepared
 import shapely.wkb
 
+class ReclassificationMissingValuesError(Exception):
+    """Raised when a raster value is not a valid key to a dictionary.
+
+    Attributes:
+        msg (str) - error message
+        missing_values (list) - a list of the missing values from the raster
+            that are not keys in the dictionary
+        value_map (dict) - the dictionary used to map raster values
```

Comment:
Hey Doug, I'm sorry I didn't notice this before until you made a note of it in your HISTORY.rst. Presumably the caller is in the same context as the exception that's raised here so the `value_map` should be in scope of handling that exception right? The exception should be providing information the caller does not have, which are the actual missing values. It's not necessarily bad to include this, but naively it seems unnecessary and just another bit of code that could be dropped. Is there a reason you're including `value_map` as an attribute in the exception?

Reply:
Nope, that's a fair assessment and does seem redundant. I'll remove it.

Reply:
Removed `value_map` here: https://github.com/natcap/pygeoprocessing/pull/106/commits/0ed7d528c4fe110c9333a5877e5d9bc00b902482